frontend/src/app/edit-claim/edit-claim.component.ts

```diff
@@ -47,22 +47,35 @@ export class EditClaimComponent implements OnInit {
     }
   }
 
-  deleteFile(filename: string): void {
-    // FIXME: This is not correct
-    this.service
-      .deleteFile(filename, this.vin)
-      .subscribe(
-        undefined,
-        error => console.log(error),
-        () => alert('FILE DELETED'),
-      );
-    // TODO: show visual confirmation of deletion
+  archiveFile(filename: string): void {
+    this.service.archiveFile(filename, this.vin).subscribe(
+      undefined,
+      error => console.log(error),
+      () => {
+        alert('FILE DELETED');
+        this.reloadComponent();
+      }
+    );
+  }
+
+  restoreFile(filename: string): void {
+    this.service.restoreFile(filename, this.vin).subscribe(
+      undefined,
+      error => console.log(error),
+      () => {
+        alert('FILE RESTORED');
+        this.reloadComponent();
+      }
+    );
   }
 
   submitChanges() {
     if (this.files) {
-      this.service.addFiles(this.files, this.vin);
-      this.reloadComponent();
+      this.service.addFiles(this.files, this.vin)
+        .subscribe(undefined, undefined, () => {
```

Comment:
Moved the subscribe to the component here as it should be - will start moving subscribes into the consumers shortly.